lib/Twig/Error.php

```diff
@@ -67,8 +69,9 @@ public function __construct($message, $lineno = -1, $name = null, Exception $pre
 
         $this->lineno = $lineno;
         $this->filename = $name;
+        $this->path = $path;
 
-        if (-1 === $lineno || null === $name) {
+        if (-1 === $lineno || null === $name || null === $path) {
```

Comment:
-1 for the null check on the path, as it would force guessing for any case where the template does not have a path because of not using a template loader relying on the filesystem. And the template guessing is a performance killer, which is why we did our best to provide the template name at compilation time for runtime errors whenever possible.